modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/containers/Form.es.js

```diff
@@ -249,29 +247,57 @@ const Form = React.forwardRef(
 
 Form.displayName = 'Form';
 
-const FormProxy = React.forwardRef(
+const FormEditor = React.forwardRef(
 	(
 		{
-			instance,
+			onEvent = () => {},
 			activePage = 0,
 			defaultLanguageId = themeDisplay.getLanguageId(),
 			...otherProps
 		},
 		ref
-	) => (
-		<FormProvider
-			onEvent={(type, payload) => instance.emit(type, payload)}
-			value={{...otherProps, activePage, defaultLanguageId}}
-		>
-			{(props) => <Form {...props} ref={ref} />}
-		</FormProvider>
-	)
+	) => {
+		const {containerId} = otherProps;
+
+		const defaultRef = useRef(null);
+
+		const reactComponentRef = ref ?? defaultRef;
+
+		useEffect(() => {
+			Liferay.component(
+				containerId,
+				{
+					reactComponentRef,
+				},
+				{
+					destroyOnNavigate: true,
+				}
+			);
```

Comment:
@matuzalemsteles @natocesarrego, does make sense adding a return closure(useEffect's unmount) here for unregister this component when this component is disposed?

Reply:
Maybe so, but either way we already have `destroyOnNavigate` set to true. This will destroy the register when browsing.